engine/src/main/java/org/camunda/bpm/engine/impl/HistoryTimeToLiveParser.java

```diff
@@ -0,0 +1,122 @@
+
+  protected static final ConfigurationLogger LOG = ConfigurationLogger.CONFIG_LOGGER;
+
+  private static final int DEFAULT_HISTORY_TIME_TO_LIVE_VALUE = 180;
+
+  private final boolean enforceNonNullValue;
+  private final String configValue;
+
+  private HistoryTimeToLiveParser(boolean enforceNonNullValue, String configValue) {
+    this.enforceNonNullValue = enforceNonNullValue;
+    this.configValue = configValue;
+  }
+
+  public static HistoryTimeToLiveParser create() {
+    ProcessEngineConfigurationImpl config = Context.getProcessEngineConfiguration();
+    Objects.requireNonNull(config, "HistoryTimeToLiveParser requires a non null config to be created");
+
+    return create(config);
+  }
+
+  public static HistoryTimeToLiveParser create(CommandContext context) {
+    ProcessEngineConfigurationImpl config = context.getProcessEngineConfiguration();
+
+    return create(config);
+  }
+
+  public static HistoryTimeToLiveParser create(ProcessEngineConfigurationImpl config) {
+    boolean enforceHistoryTimeToLive = config.isEnforceHistoryTimeToLive();
+    String historyTimeToLive = config.getHistoryTimeToLive();
+
+    return new HistoryTimeToLiveParser(enforceHistoryTimeToLive, historyTimeToLive);
+  }
+
+  public void validate(Integer historyTimeToLive) {
+    if (enforceNonNullValue && historyTimeToLive == null) {
+      throw new NotAllowedException("Null historyTimeToLive values are not allowed");
+    }
+  }
+
+  public Integer parse(Element processElement) {
+    String historyTimeToLiveString = processElement.attributeNS(CAMUNDA_BPMN_EXTENSIONS_NS, "historyTimeToLive", configValue);
+
+    return parseAndValidate(historyTimeToLiveString);
+  }
+
+  public Integer parse(Case caseElement) {
+    String historyTimeToLiveString = getValueOrConfig(caseElement.getCamundaHistoryTimeToLiveString());
+
+    return parseAndValidate(historyTimeToLiveString);
+  }
+
+  public Integer parse(Decision decision) {
+    String historyTimeToLiveString = getValueOrConfig(decision.getCamundaHistoryTimeToLiveString());
+
+    return parseAndValidate(historyTimeToLiveString);
+  }
+
+  /**
+   * Parses the given HistoryTimeToLive String expression and then executes any applicable validation before returning
+   * the parsed value.
+   *
+   * @param historyTimeToLiveString the history time to live string expression in ISO-8601 format
+   * @return the parsed integer value of history time to live
+   * @throws NotValidException in case enforcement of non-null values is on and the parsed result was null
+   */
+  protected Integer parseAndValidate(String historyTimeToLiveString) throws NotValidException {
+    Integer result = ParseUtil.parseHistoryTimeToLive(historyTimeToLiveString);
+
+    if (enforceNonNullValue && result == null) {
+      throw new NotValidException("History Time To Live cannot be null");
+    }
+
+    if (result != null && result == DEFAULT_HISTORY_TIME_TO_LIVE_VALUE) {
+      LOG.logHistoryTimeToLiveDefaultValueWarning();
```

Comment:
🔧 Can we pass the process definition id to the logger and log it? 🤔 Like this, users know later on which processes are affected.

Reply:
The processDefinitionId now is passed and will be present only when parsing processes and cases. For Decisions that can be deployed separately, it is passed as null and eventually ignored during the logging. Does this make sense?